cmd/snap/cmd_run.go

```diff
@@ -217,7 +240,11 @@ func runSnapConfine(info *snap.Info, securityTag, snapApp, command, hook string,
 	cmd = append(cmd, snapApp)
 	cmd = append(cmd, args...)
 
-	env := append(os.Environ(), snapExecEnv(info)...)
+	env := envMap(os.Environ())
+	snapEnv := envMap(snapExecEnv(info))
+	for k, v := range snapEnv {
```

Comment:
I'd add a comment that this explicitly overrides all of the values from `snapExecEnv(...)`